src/components/data/DataTable/TableRow.tsx

```diff
@@ -17,38 +57,27 @@ import TableDataCell from './TableDataCell';
  */
 
 export default function TableRow({
-    // @ts-expect-error TS(7031) FIXME: Binding element 'id' implicitly has an 'any' type.
     id,
-    // @ts-expect-error TS(7031) FIXME: Binding element 'children' implicitly has an 'any'... Remove this comment to see the full error message
     children,
-    // @ts-expect-error TS(7031) FIXME: Binding element 'selected' implicitly has an 'any'... Remove this comment to see the full error message
     selected,
-    // @ts-expect-error TS(7031) FIXME: Binding element 'onClick' implicitly has an 'any' ... Remove this comment to see the full error message
     onClick,
-    // @ts-expect-error TS(7031) FIXME: Binding element 'onMouseOver' implicitly has an 'a... Remove this comment to see the full error message
     onMouseOver,
-    // @ts-expect-error TS(7031) FIXME: Binding element 'onMouseOut' implicitly has an 'an... Remove this comment to see the full error message
     onMouseOut,
-    // @ts-expect-error TS(7031) FIXME: Binding element 'showCols' implicitly has an 'any'... Remove this comment to see the full error message
     showCols,
-    // @ts-expect-error TS(7031) FIXME: Binding element 'className' implicitly has an 'any... Remove this comment to see the full error message
     className,
-    // @ts-expect-error TS(7031) FIXME: Binding element 'style' implicitly has an 'any' ty... Remove this comment to see the full error message
     style
-}) {
-    // @ts-expect-error TS(7006) FIXME: Parameter 'index' implicitly has an 'any' type.
-    const showData = index => (index < showCols.length ? showCols[index] : true);
+}: PropsWithChildren<TableRowProps & TRProps>) {
+    const showData = (index: number) => (index < showCols.length ? showCols[index] : true);
     const computedClassName = className + (selected ? ' active' : '');
-    // @ts-expect-error TS(7034) FIXME: Variable 'computedChildren' implicitly has type 'a... Remove this comment to see the full error message
-    const computedChildren = [];
+    const computedChildren: ReactNode[] = [];
 
     const computedStyle = { ...style };
     if (_.isFunction(onClick)) {
         computedStyle.cursor = 'pointer';
     }
 
     React.Children.forEach(children, (child, index) => {
-        if (child && child.type === TableDataCell && showData(index)) {
+        if (child && typeof child === 'object' && 'type' in child && child.type === TableDataCell && showData(index)) {
```

Comment:
WDYT about the following simplification:

```suggestion
        if (child && child?.type === TableDataCell && showData(index)) {
```

? 🤔

Reply:
typescript doesnt understand this, sadly, i will try again though

Reply:
I didn't double check it but maybe try: `if (child && (child as ReactElement).type === TableDataCell && showData(index)) {` maybe that would solve the issue? ofc, we would need to import type ReactElement from React as well

Reply:
typically I avoid "as", but here it is excellent choice, thank you